### PR TITLE
Add `chainerx.flatten`

### PR DESCRIPTION
**Pull request by aksub99**
Add `chainerx.flatten`

### CODE REVIEW

File: chainerx_cc/chainerx/routines/manipulation.cc [hunk 1]
```diff
@@ -737,6 +737,11 @@ Array ExpandDims(const Array& a, int8_t axis) {
     return out;
 }
 
+Array Flatten(const Array& a) {
+    Array out = a.Reshape({a.GetTotalSize()});
```
Comment (asi1024) [inline chainerx_cc/chainerx/routines/manipulation.cc]: `numpy.ndarray.flatten` always return their copy.
```suggestion
    Array out = a.Copy().Reshape({a.GetTotalSize()});
```

File: chainerx_cc/chainerx/python/routines.cc [hunk 2]
```diff
@@ -547,6 +547,7 @@ void InitChainerxManipulation(pybind11::module& m) {
           [](const ArrayBodyPtr& a, int8_t axes) { return MoveArrayBody(Transpose(Array{a}, {axes})); },
           "a"_a,
           "axes"_a = nullptr);
+    m.def("flatten", [](const ArrayBodyPtr& a) { return MoveArrayBody(Flatten(Array{a})); }, "a"_a);
```
Comment (asi1024) [inline chainerx_cc/chainerx/python/routines.cc]: Move the definition to `chainerx_cc/chainerx/python/array.cc`.

File: chainerx_cc/chainerx/routines/manipulation.cc (hunk 1 above)
Comment (aksub99) [reply chainerx_cc/chainerx/routines/manipulation.cc]: Fixed.

File: chainerx_cc/chainerx/python/routines.cc (hunk 2 above)
Comment (aksub99) [reply chainerx_cc/chainerx/python/routines.cc]: Done.

**Comment by aksub99** [comment]
I have made the required changes.

File: chainerx_cc/chainerx/routines/manipulation.h [hunk 3]
```diff
@@ -61,6 +61,8 @@ Array Repeat(const Array& a, const std::vector<int64_t>& repeats, absl::optional
 
 Array ExpandDims(const Array& a, int8_t axis);
 
+Array Flatten(const Array& a);
```
Comment (asi1024) [inline chainerx_cc/chainerx/routines/manipulation.h]: Remove this declaration.

Comment (aksub99) [reply chainerx_cc/chainerx/routines/manipulation.h]: Done. Sorry, missed that.

File: chainerx_cc/chainerx/routines/manipulation.cc (hunk 1 above)
Comment (asi1024) [reply chainerx_cc/chainerx/routines/manipulation.cc]: Why didn't you apply this fix? The code in my suggestion seems simpler.

**Comment by asi1024** [comment]
Could you fix for cpplint?

File: chainerx_cc/chainerx/routines/manipulation.cc (hunk 1 above)
Comment (aksub99) [reply chainerx_cc/chainerx/routines/manipulation.cc]: Fixed now. Please see if it's alright now.

**Comment by aksub99** [comment]
I split the routines but it looks like there's a linker error due to mismatch in declaration and definition. Is there some other file I have to modify or should I check if I have mistakenly modified some functions?

**Comment by asi1024** [comment]
`InitChainerxArray` is declared but not implemented.

**Comment by aksub99** [comment]
I have modified the declaration. I still get the same error.

**Comment by aksub99** [comment]
I have split the functions. I get the error:
```
ImportError while loading conftest '/home/akshay/chainer/tests/chainerx_tests/conftest.py'.
tests/chainerx_tests/conftest.py:3: in <module>
    import chainerx.testing
chainerx/__init__.py:75: in <module>
    _docs.set_docs()
chainerx/_docs/__init__.py:24: in set_docs
    m.set_docs()
chainerx/_docs/array.py:172: in set_docs
    ndarray.argmax,
E   AttributeError: type object 'chainerx.ndarray' has no attribute 'argmax'
```
Please review.

File: chainerx_cc/chainerx/python/array.cc [hunk 4]
```diff
@@ -468,6 +467,9 @@ void InitChainerxArray(pybind11::module& m) {
           py::is_operator());
     c.def("__rshift__", [](const ArrayBodyPtr& self, Scalar rhs) { return MoveArrayBody(Array{self} >> rhs); }, py::is_operator());
     c.def("__rrshift__", [](const ArrayBodyPtr& self, Scalar lhs) { return MoveArrayBody(lhs >> Array{self}); }, py::is_operator());
+}
+
+void InitChainerxCalculation(py::class_<ArrayBody, ArrayBodyPtr> c) {
```
Comment (asi1024) [inline chainerx_cc/chainerx/python/array.cc]: Rename this function and call it from `InitChainerxArray`.
```suggestion
void InitChainerxArrayCalculation(py::class_<ArrayBody, ArrayBodyPtr> c) {
```

Comment (aksub99) [reply chainerx_cc/chainerx/python/array.cc]: Made all the changes. Tests pass successfully now. Thanks for the review.

**Comment by asi1024** [comment]
pfnCI, test this please.

**Comment by asi1024** [comment]
LGTM. Thank you for the PR!

**Pull request merged**